### DISCUSSION TITLE
Misidentifying a (some?) Java repo?

### DISCUSSION THREAD

**Discussion by delabassee** [github-linguist/linguist#6255] [Classification] [answered]
## Describe the bug

This [public repo](https://github.com/ShehanPerera/javaagent-bytebuddy) contains 
- 18 Java sources with over 400 LoC in total
- 7 Shell scripts with 14 LoC in total

and yet, the repo is reported as "100% Shell"

```
github-linguist
100.00% 1031       Shell
```

```
github-linguist --breakdown
100.00% 1031       Shell

Shell:
run-add-field.sh
run-add-method.sh
run-constructor-instrument.sh
run-execution-time.sh
run-get-methods.sh
run-get-parameters.sh
run-ignore-class.sh
```

### Expected behaviour
I would expect Java to be reported as the main language for this repo.

### Related discussion
NA

**Comment by lildude** [comment, marked as answer]
This is happening because all the Java files are under the `samples` directory which Linguist considers as documentation:

https://github.com/github/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/lib/linguist/documentation.yml#L33-L34

You will need to implement a `-linguist-documentation` override or rename the directory.

**Comment by delabassee** [comment]
Shouldn't this be addressed? In the end, the inferred language (shell in this case) isn't correct.

**Reply by lildude** [reply]
Not from Linguist. Many projects use `samples` or similar named directories to hold samples used for documentation purposes. If a repo owner doesn't like this, they need to implement the override suggested. If you don't like it, reach out to the repo owner to add the necessary override.

I should also point out that even if we removed this from Linguist, this repo wouldn't be immediately affected because we only analyse repos when there is a push (we don't retrospectively scan repos as this would be prohibitively costly in terms of unnecessary resource usage) and this repo hasn't been touched in 4 years.
